src/store/proceedings.ts

```diff
@@ -0,0 +1,85 @@
+import type { Request } from '../types/request';
+import { Message, MessageId, Proceeding } from '../types/proceedings.d';
+import create from 'zustand';
+import { devtools, persist } from 'zustand/middleware';
+import { produce } from 'immer';
+import { Privacy, PRIVACY_ACTIONS } from '../Utility/Privacy';
+
+export interface ProceedingsState {
+    proceedings: Record<string, Proceeding>;
+    addProceeding: (proceeding: Proceeding) => void;
+    addMessage: (message: Message) => void;
+    removeMessage: (id: MessageId) => void;
+    addAttachment: (id: MessageId, file: any) => void;
+    removeProceeding: (reference: string) => void;
+    clearProceedings: () => void;
+}
+
+const id_regex = /^(\d*?-[\dA-Za-z]*?)-(\d*)$/;
+
+export const useProceedingsStore = create<ProceedingsState>(
+    devtools(
+        persist(
+            (set, get) => ({
+                proceedings: {},
+                addProceeding: (proceeding) =>
+                    Privacy.isAllowed(PRIVACY_ACTIONS.SAVE_MY_REQUESTS) &&
```

Comment:
Would you wrap the localforage instance with this?

Reply:
Good idea. Considering we'll need more than one of these persisted stores, it might even make sense to create a localforage wrapper that only does something if some condition is met.